src/App.tsx

```diff
@@ -1,56 +1,91 @@
-import React from 'react';
-import './App.scss';
-import { peopleFromServer } from './data/people';
+/* eslint-disable @typescript-eslint/quotes */
+import React, { useCallback, useMemo, useState } from "react";
+import "./App.scss";
+import classNames from "classnames";
+import debounce from "lodash/debounce";
+import { peopleFromServer } from "./data/people";
+import { Person } from "./types/Person";
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState("");
+  const [appliedQuery, setAppliedQuery] = useState("");
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [showMenu, setShowMenu] = useState(false);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const applyQuery = useCallback(debounce(setAppliedQuery, 500), []);
+
+  const filteredPeople = useMemo(
+    () => peopleFromServer.filter(
+      person => person.name.toLowerCase().includes(appliedQuery.toLowerCase()),
+    ), [appliedQuery],
+  );
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (appliedQuery.length === 0) {
```

Comment:
means when user start typing text dropdown should hide and apear after delay
```suggestion
   setShowMenu(false)
    if (appliedQuery.length === 0) {
```